input/script_utils.go

```diff
@@ -702,23 +709,29 @@ func SenderHTLCScriptTaprootRedeem(signer Signer, signDesc *SignDescriptor,
 
 	// In addition to the signature and the witness/leaf script, we also
 	// need to make a control block proof using the tapscript tree.
-	successTapLeafHash := txscript.NewBaseTapLeaf(
-		signDesc.WitnessScript,
-	).TapHash()
-	successIdx := tapscriptTree.LeafProofIndex[successTapLeafHash]
-	successMerkleProof := tapscriptTree.LeafMerkleProofs[successIdx]
-	successControlBlock := successMerkleProof.ToControlBlock(revokeKey)
+	var ctrlBlock []byte
+	if signDesc.ControlBlock == nil {
```

Comment:
With the current control flow, since we write the sign desc to disk in the resolvers, and the control block for the sign desc is populated when making the resolutions, it's always set. For tests, it was easier to not have to worry about making the control block, instead letting this routine make it itself.

Reply:
Imo if it's just for testing then we should create it in the test. If the `ControlBlock` is expected to not be nil here, yet it's nil, then maybe we've caught a bug we need to fix.

Reply:
>  If the ControlBlock is expected to not be nil here, yet it's nil, then maybe we've caught a bug we need to fix

Yes, that's why all those checks were added in the first place: they revealed bugs where the control block shouldn't have been nil. Many of those cases were due to not properly writing the necessary information to disk. So I see the checks more as belt and suspenders checks: we'll error out rather than panic the entire daemon.